lib/team_statistics.rb

```diff
@@ -0,0 +1,109 @@
+module TeamStatistics
+
+  def team_info(team_id)
+    found_team = @teams.fetch(team_id.to_i)
+    {
+      "team_id" => found_team.team_id.to_s,
+      "franchise_id" => found_team.franchise_id.to_s,
+      "team_name" => found_team.team_name,
+      "abbreviation" => found_team.abbreviation,
+      "link" => found_team.link
+    }
+  end
+
+  def best_season(team_id)
+    team_id = team_id.to_i
+    filtered_games = []
+    games_by_team = @games.each do |game_id, game|
+      if game.home_team_id == team_id || game.away_team_id == team_id
+        filtered_games.push(game)
+      end
+    end
+
+    games_by_season = filtered_games.group_by do |game|
+      game.season
+    end
+
+    season_win_avg = Hash.new(0)
+    games_by_season.each do |season, games|
+      home_wins = games.find_all do |game|
+        game.home_team_id == team_id && game.home_goals > game.away_goals
+      end
+
+      away_wins = games.find_all do |game|
+        game.away_team_id == team_id && game.home_goals < game.away_goals
+      end
+      season_win_avg[season] = ((home_wins.length.to_f + away_wins.length) / games.length).round(2)
+    end
+    season_win_avg.key(season_win_avg.values.max_by { |value| value})
+  end
+
+  def worst_season(team_id)
+    team_id = team_id.to_i
+    filtered_games = []
+    games_by_team = @games.each do |game_id, game|
```

Comment:
Like above, we aren't calling the games_by_team variable anywhere in the worst_season method so we can remove it.

Reply:
@lrs8810, yes good point! I will make the change!